src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
Thus far this PR only adds potential functionality through the `HardwareJointLimitValidator` class but does not change the logic of anything else.
The issue is that changing the logic as suggested here will cause some `prpy` unit tests to fail. A unit test for checking invalid limits expects an error to be thrown, but under this new API it will just temporarily extend the joint limits rather than throw an error. Should we edit those unit tests @mkoval ?